### PR TITLE
Docker container for ci test

### PR DESCRIPTION
**Pull request by dreamer2368**
By using docker container, github ci test can contain only `libROM`-related parts for testing. Most installation steps for dependencies can be skipped, which can reduce the linux ci test time from about 15 mins to 5 mins.
- The recipe for the docker container is stored as `.github/workflows/Dockerfile`.
- The following steps are deleted, because all dependencies including `googletest` are pre-installed in the container:
  - `Install Linux Dependencies`
  - `./.github/workflows/checkout_repo` 
  - `run: scripts/setup.sh`
- `Set Swap Space` step is not needed any longer, because all dependencies are not installed within ci test. However, this step is still included in the CI test, just in case we may need extra space in the future.

Note:
- The docker container keeps the specific version (or commit) of `mfem`. While it necessitates manually updating the container occasionally, it also enables a stable CI test and development.

Issue:
- Mac CI test has been failing before this pull request, and this issue is separately handled by the pull request #175. Removed from the ci workflow .
- `RandomizedSVDTest` is susceptible to round-off error, failing randomly on different machine. This affects the test result under the container. See Issue #193 . Commented the corresponding test until the issue is resolved by a separate pull request.

TODO:
- Transfer the docker container [dreamer2368/librom_env](https://hub.docker.com/repository/docker/dreamer2368/librom_env/general) to [LLNL dockerhub](https://hub.docker.com/u/llnl)?

### CODE REVIEW

**Review by chldkdtn — APPROVED** [review]

**Comment by chldkdtn** [comment]
@debog @dylan-copeland @siuwuncheung Would you mind reviewing this PR?

File: .github/workflows/ci.yml
Comment (sullan2) [inline .github/workflows/ci.yml:3]: Can we add a workflow_dispatch trigger for the CI workflow so we have the ability to manually run the checks without a PR

**Review by sullan2 — APPROVED** [review]

File: .github/workflows/run_tests/action.yml
Comment (siuwuncheung) [inline .github/workflows/run_tests/action.yml:21]: Do we want to resolve issue #193 in this PR or open up a new PR?

**Comment by debog** [comment]
> @debog @dylan-copeland @siuwuncheung @sullan2 Would you mind reviewing this PR?

I think I am out of my depths here!

File: .github/workflows/ci.yml
Comment (dreamer2368) [reply .github/workflows/ci.yml:3]: @sullan2 , I saw `macos_fix` branch has that trigger in the ci workflow. Should we wait until that branch is merged or reflect it right away here in this branch?

File: .github/workflows/ci.yml
Comment (dreamer2368) [reply .github/workflows/ci.yml:65]: @siuwuncheung , we know the issue and the mac workflow is separate from the linux one so test is still effective. But since this is a public repo which many other people use, for visibility to them, I **personally** think it would be better to comment this out, until the mac ci workflow is actually established. But we can uncomment this if we think the conflict is more of the issue.

File: .github/workflows/run_tests/action.yml
Comment (dreamer2368) [reply .github/workflows/run_tests/action.yml:21]: I think #193 is a separate issue so I propose to open up a new PR.

File: .github/workflows/ci.yml
Comment (sullan2) [reply .github/workflows/ci.yml:3]: @dreamer2368 It would be good to have the trigger now, for easier testing.

File: .github/workflows/ci.yml
Comment (siuwuncheung) [reply .github/workflows/ci.yml:65]: Sounds good to me. I'll approve the PR right away.

**Comment by dylan-copeland** [comment]
> > @debog @dylan-copeland @siuwuncheung @sullan2 Would you mind reviewing this PR?
> 
> I think I am out of my depths here!

Me too, and there are enough approvals already, so I will not review.

**Pull request merged**
